node/src/components/contract_runtime/operations.rs

```diff
@@ -402,16 +402,23 @@ where
 }
 
 /// Computes the root hash for a Merkle tree constructed from the hashes of execution results.
+///
+/// NOTE: We're hashing vector of execution results, instead of just their hashes, b/c when a joiner
+/// node receives the chunks of *full data* it has to be able to verify it against the merkle root.
 fn compute_execution_results_root_hash<'a>(
     results: &mut impl Iterator<Item = &'a ExecutionResult>,
 ) -> Result<Digest, BlockCreationError> {
-    let mut execution_results = vec![];
+    let mut execution_results_serialized = vec![];
```

Comment:
I think we need to do something more like changing `results` to a `Vec<ExecutionResult>` and just call `let execution_results_bytes = results.to_bytes()...` - that way we get the length of the vector written to the output bytes first, which is what I think we'll need when serializing the actual execution results to send them.

Reply:
Fixed in [`51e7d25` (#3263)](https://github.com/casper-network/casper-node/pull/3263/commits/51e7d2555cbe6230bf85db1b3ce073378eac6cc5).